tests/ci-install.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+set -euo pipefail
+set -x
+
+NULL=
+: "${ci_docker:=}"
+: "${ci_in_docker:=}"
+: "${ci_suite:=jessie}"
+
+if [ $(id -u) = 0 ]; then
+    sudo=
+else
+    sudo=sudo
+fi
+
+if [ -n "$ci_docker" ]; then
+    sed \
+        -e "s/@ci_distro@/${ci_distro}/" \
+        -e "s/@ci_docker@/${ci_docker}/" \
+        -e "s/@ci_suite@/${ci_suite}/" \
+        < tests/ci-Dockerfile.in > Dockerfile
+    exec docker build -t ostree-ci .
+fi
+
+case "$ci_distro" in
+    (debian)
+        # httpredir.debian.org seems to be unreliable, use a CDN instead
+        sed -i -e 's/httpredir\.debian\.org/deb.debian.org/g' /etc/apt/sources.list
+        ;;
+esac
+
+case "$ci_suite" in
+    (jessie)
+        $sudo apt-get -y update
+        $sudo apt-get -y install apt-transport-https wget
+        wget -O - https://sdk.gnome.org/apt/debian/conf/alexl.gpg.key | $sudo apt-key add -
+        echo "deb [arch=amd64] https://sdk.gnome.org/apt/debian/ jessie main" | $sudo tee /etc/apt/sources.list.d/flatpak.list
+        ;;
+
+    (trusty|xenial)
+        $sudo apt-get -y update
+        $sudo apt-get -y install software-properties-common
+        $sudo add-apt-repository --yes ppa:alexlarsson/flatpak
+        ;;
+esac
+
+case "$ci_distro" in
+    (debian|ubuntu)
+        $sudo apt-get -y update
+        $sudo apt-get -y install \
+            attr \
+            bison \
+            cpio \
+            debhelper \
+            dh-autoreconf \
+            dh-systemd \
+            docbook-xml \
+            docbook-xsl \
+            e2fslibs-dev \
+            elfutils \
+            fuse \
+            gjs \
+            gnome-desktop-testing \
+            gobject-introspection \
+            gtk-doc-tools \
+            libarchive-dev \
+            libattr1-dev \
+            libcap-dev \
+            libfuse-dev \
+            libgirepository1.0-dev \
+            libglib2.0-dev \
+            libgpgme11-dev \
+            libgsystem-dev \
+            liblzma-dev \
+            libmount-dev \
+            libselinux1-dev \
+            libsoup2.4-dev \
+            procps \
+            zlib1g-dev \
+            ${NULL}
+
+        if [ -n "$ci_in_docker" ]; then
+            adduser --disabled-password user </dev/null
+            apt-get -y install sudo systemd-sysv
+            echo "user ALL=(ALL) NOPASSWD: ALL" > /etc/sudoers.d/nopasswd
+            chmod 0440 /etc/sudoers.d/nopasswd
+        fi
+        ;;
+
+    (*)
```

Comment:
I am very strongly of the opinion that CI/packaging should be tightly integrated, and one should avoid things like hardcoded lists of package names in CI that duplicate the downstream deps.   So I maintain https://github.com/cgwalters/rpmdistro-gitoverlay which solves that problem for the Fedora/CentOS world.

(Right now our internal CI isn't using it, but I hope to convert)